request.go

```diff
@@ -28,9 +29,18 @@ type (
 	// ExtraQueryValidationError the key used in `Context.Extra` to
 	// store the query validation errors.
 	ExtraQueryValidationError struct{}
+
+	// ExtraRequestError the key used in `Context.Extra` to
+	// store specific request errors.
+	ExtraRequestError struct{}
+)
+
+var (
+	// ErrInvalidJSONBody error for invalid JSON body
+	ErrInvalidJSONBody = errors.New("request.json-invalid-body")
```

Comment:
If we are using errors, we should wrap them with the actual error and use `errors.Is` in the status handler to get the corresponding message. I think it would be better.


Simplified example:
```go
r.Extra[goyave.ExtraParseError{}] = fmt.Errorf("%w: %w", goyave.ErrInvalidJSONBody, err)
```

```go
err := request.Extra[ExtraRequestError{}]
switch {
    case errors.Is(err, ErrInvalidJSONBody):
        message := lang.Get("parse.json-invalid-body")
    //...
}

Reply:
The fist part yes, but why the second part? Isn't that limiting ourselves?

Reply:
I think it is better from a responsibility point of view. We get an error and the status handler renders it. Putting the lang entry as the error message is mixing those concepts together. On top of that, using the real error as a source gives more flexibility to the status handler for rendering it as it contains more details, should we use them one day in the future.

Reply:
Done.